src/parsers/power_system_table_data.jl

```diff
@@ -815,54 +815,38 @@ function calculate_variable_cost(
 
-    vom = isnothing(gen.variable_cost) ? 0.0 : gen.variable_cost
+    isnothing(gen.variable_cost) ||
+        error(
+            "Cannot combine non-fuel-based `variable_cost` with fuel-based cost currently being parsed",
+        )
 
-    if fuel_cost > 0.0
-        var_cost =
-            [(getfield(gen, hr), getfield(gen, mw)) for (hr, mw) in cost_colnames.columns]
-        var_cost = unique([
-            (tryparse(Float64, string(c[1])), tryparse(Float64, string(c[2]))) for
-            c in var_cost if !in(nothing, c)
-        ])
-        if isempty(var_cost)
-            @warn "Unable to calculate variable cost for $(gen.name)" var_cost maxlog = 5
-        end
-    else
-        var_cost = [(0.0, 0.0)]
-    end
+    var_cost =
+        [(getfield(gen, hr), getfield(gen, mw)) for (hr, mw) in cost_colnames.columns]
+    var_cost = unique([
+        (tryparse(Float64, string(c[1])), tryparse(Float64, string(c[2]))) for
+        c in var_cost if !in(nothing, c)
+    ])
+    var_cost = [
+        (
+            (var_cost[i][1]),
+            (var_cost[i][2] .* gen.active_power_limits_max .* base_power),
+        ) for i in 1:length(var_cost)
+    ]
+    isempty(var_cost) &&
+        @warn "Unable to calculate variable cost for $(gen.name)" var_cost maxlog = 5
 
     if length(var_cost) > 1
-        var_cost[2:end] = [
-            (
-                (
-                    var_cost[i][1] * fuel_cost * (var_cost[i][2] - var_cost[i - 1][2]) +
-                    var_cost[i][2] * vom
-                ),
-                var_cost[i][2],
-            ) .* gen.active_power_limits_max .* base_power for i in 2:length(var_cost)
-        ]
-        var_cost[1] =
-            ((var_cost[1][1] * fuel_cost + vom) * var_cost[1][2], var_cost[1][2]) .*
-            gen.active_power_limits_max .* base_power
-
-        fixed = max(
-            0.0,
-            var_cost[1][1] -
-            (var_cost[2][1] / (var_cost[2][2] - var_cost[1][2]) * var_cost[1][2]),
+        initial_rate, initial_output = var_cost[1]
+        var_cost = IncrementalCurve(
+            PiecewiseLinearData([(x, y) for (y, x) in var_cost[2:end]]),
+            initial_rate * initial_output,
         )
-        var_cost[1] = (var_cost[1][1] - fixed, var_cost[1][2])
-
-        for i in 2:length(var_cost)
-            var_cost[i] = (var_cost[i - 1][1] + var_cost[i][1], var_cost[i][2])
-        end
-        var_cost = PiecewiseLinearData([(x, y) for (y, x) in var_cost])
-
     elseif length(var_cost) == 1
-        # if there is only one point, use it to determine the constant $/MW cost
-        var_cost = LinearFunctionData(var_cost[1][1] * fuel_cost + vom)
-        fixed = 0.0
+        # if there is only one point, use it to determine the constant fuel/MW cost
+        var_cost = AverageRateCurve(LinearFunctionData(0, var_cost[1][1]), 0.0)
     end
-    return var_cost, fixed, fuel_cost
+    # @show FuelCurve(var_cost, fuel_cost)  # TODO test correctness of this conversion
```

Comment:
is this still pending?

Reply:
Yeah and `test_power_system_table_data.jl` is suggesting that I did the math wrong in the other section too

Reply:
This is on somewhat firmer ground now but I'm putting it in the TODO issue